#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From a14db20143617583a2b9fac87833c583b1dafb44 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marc-Andr=C3=A9=20Lureau?= <marcandre.lureau@gmail.com>
Date: Wed, 20 Mar 2013 23:12:56 +0100
Subject: [PATCH] win32: do not crash on invalid utf8 conversion

g_utf8_to_utf16() is not guaranteed to succeed. Check the error
and return if it failed.

https://bugzilla.gnome.org/show_bug.cgi?id=696232
---
 gdk/win32/gdkproperty-win32.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/gdk/win32/gdkproperty-win32.c b/gdk/win32/gdkproperty-win32.c
index 9096729..1e47d4b 100644
--- a/gdk/win32/gdkproperty-win32.c
+++ b/gdk/win32/gdkproperty-win32.c
@@ -158,6 +158,7 @@ gdk_property_change (GdkWindow    *window,
   guchar *ucptr;
   wchar_t *wcptr, *p;
   glong wclen;
+  GError *err = NULL;

   g_return_if_fail (window != NULL);
   g_return_if_fail (GDK_IS_WINDOW (window));
@@ -201,7 +202,13 @@ gdk_property_change (GdkWindow    *window,
	      return;
	    }

-	  wcptr = g_utf8_to_utf16 ((char *) data, nelements, NULL, &wclen, NULL);
+	  wcptr = g_utf8_to_utf16 ((char *) data, nelements, NULL, &wclen, &err);
+          if (err != NULL)
+            {
+              g_warning ("Failed to convert utf8: %s", err->message);
+              g_clear_error (&err);
+              return;
+            }

	  wclen++;		/* Terminating 0 */
	  size = wclen * 2;
--
1.8.1.1.439.g50a6b54
